pkg/installer/upgrade.go

```diff
@@ -58,6 +74,29 @@ func Upgrade(uninstallConfig *apiv1.KubectlStorageOSConfig, installConfig *apiv1
 	return err
 }
 
+// checkForExistingWorkloads ensures no bound PVCs are using a storageos storageclass.
+func (in *Installer) checkForExistingWorkloads() error {
+	pvcList, err := pluginutils.ListPersistentVolumeClaims(in.clientConfig, metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	stosSCList, err := pluginutils.ListStorageClasses(in.clientConfig, metav1.ListOptions{LabelSelector: stosAppLabel})
+	if err != nil {
+		return err
+	}
+	for _, pvc := range pvcList.Items {
+		if pvc.Status.Phase != corev1.ClaimBound {
+			continue
+		}
+		for _, stosSC := range stosSCList.Items {
+			if *pvc.Spec.StorageClassName == stosSC.Name {
+				return fmt.Errorf(errWorkloadsExist, pvc.Name, *pvc.Spec.StorageClassName)
+			}
```

Comment:
I think you need to check for default StorageClasses too.  

Check https://github.com/storageos/api-manager/blob/master/internal/pkg/provisioner/storageos.go#L85-L95.  There are a few other functions in that package that might also be useful.  You may be able to use `HasStorageOSAnnotation(obj client.Object) bool`